book-content/chapters/06-objects.md

```diff
@@ -0,0 +1,1386 @@
+  price: number;
+  description: string;
+}
+
+const updateProduct = (id: number, productInfo: Product) => {
+  // Do something with the productInfo
+};
+```
+
+The twist here is that during a product update, we might not want to modify all of its properties at the same time. Because of this, not all properties have to be passed into the function.
+
+This means we have several different test scenarios. For example, update just the name, just the price, or just the description. Combinations like updating the name and the price or the name and the description are also tested.
+
+```ts twoslash
+// @errors: 2345
+interface Product {
+  id: number;
+  name: string;
+  price: number;
+  description: string;
+}
+
+const updateProduct = (id: number, productInfo: Product) => {
+  // Do something with the productInfo
+};
+
+// ---cut---
+updateProduct(1, {
+  name: "Book",
+});
+
+updateProduct(1, {
+  price: 12.99,
+});
+```
+
+Your challenge is to modify the `productInfo` parameter to reflect these requirements. The `id` should remain absent from `productInfo`, but we also want all other properties in `productInfo` to be optional.
+
+<Exercise title="Exercise 2: Updating a Product" filePath="/src/020-objects/091-omit-type-helper.problem.ts"></Exercise>
+
+#### Solution 1: Expecting Certain Properties
+
+There are quite a few ways to solve this problem. Here are a few examples:
+
+##### Using Pick
+
+Using the Pick utility type, we can create a new type that only includes the `name` and `email` properties from the `User` interface:
+
+```typescript
+type PickedUser = Pick<User, "name" | "email">;
+```
+
+Then the `fetchUser` function can be updated to return a `Promise` of `PickedUser`:
+
+```typescript
+const fetchUser = async (): Promise<PickedUser> => {
+  ...
+```
+
+##### Using Omit
+
+The Omit utility type can also be used to create a new type that excludes the `id` and `role` properties from the `User` interface:
+
+```typescript
+type OmittedUser = Omit<User, "id" | "role">;
+```
+
+Then the `fetchUser` function can be updated to return a `Promise` of `OmittedUser`:
+
+```typescript
+const fetchUser = async (): Promise<OmittedUser> => {
+  ...
+```
+
+##### Extending an Interface
+
+We could create an interface `NameAndEmail` that contains a `name` and `email` property, along with updating the `User` interface to remove those properties in favor of extending them:
+
+```typescript
+interface NameAndEmail {
+  name: string;
+  email: string;
+}
+
+interface User extends NameAndEmail {
+  id: string;
+  role: string;
+}
+```
+
+Then the `fetchUser` function could return a `Promise` of `NameAndEmail`:
+
+```typescript
+const fetchUser = async (): Promise<NameAndEmail> => {
+  // ...
+};
+```
+
+`Omit` will mean that the object grows as the source object grows. `Pick` and `interface extends` will mean that the object will stay the same size. So depending on requirements, you can choose the best approach.
+
+#### Solution 2: Updating a Product
```

Comment:
Usually, your Solution X were interchangeable but, in here, this one feels like an extension of the previous one (like an extra step). The previous one doesn't mention `Partial` at all (which is a requirement here).

Reply:
hm, maybe this just isn't true? I see now that Solution X is meant to correspond to Excercise X. I was under impression that before you were using multiple Solutions to a single Excercise at times but maybe those were always presented under subheaders or smth